File: test/test_unit_symbol_invert.rb
require_relative "../lib/rgfa.rb"
require "test/unit"

TestUnit ||= Module.new

class TestUnit::SymbolInvert < Test::Unit::TestCase

  def test_invert_orientations
    assert_equal(:+, :-.invert)
    assert_equal(:-, :+.invert)
  end

  def test_invert_segment_ends
    assert_equal(:L, :R.invert)
    assert_equal(:R, :L.invert)
  end

  def test_invert_invalid
    assert_raise(RGFA::ValueError) { :xx.invert }
  end

end

File: tests/test_unit_symbol_invert.py
import unittest
import gfapy

class TestUnitSymbolInvert(unittest.TestCase):

  def test_invert_orientations(self):
    self.assertEqual("+", gfapy.invert("-"))
    self.assertEqual("-", gfapy.invert("+"))

  def test_invert_segment_ends(self):
    self.assertEqual("L", gfapy.invert("R"))
    self.assertEqual("R", gfapy.invert("L"))

  def test_invert_invalid(self):
    self.assertRaises(gfapy.ValueError, gfapy.invert, "xx")

